nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RecordStats.java

```diff
@@ -0,0 +1,165 @@
+    public Set<Relationship> getRelationships() {
+        return new HashSet<Relationship>() {{
+            add(REL_SUCCESS);
+            add(REL_FAILURE);
+        }};
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile input = session.get();
+        if (input == null) {
+            return;
+        }
+
+        try {
+            Map<String, RecordPath> paths = getRecordPaths(context);
+            Map<String, String> stats = getStats(input, paths, context, session);
+
+            input = session.putAllAttributes(input, stats);
+
+            session.transfer(input, REL_SUCCESS);
+
+        } catch (Exception ex) {
+            getLogger().error("Error processing stats.", ex);
+            session.transfer(input, REL_FAILURE);
+        }
+
+    }
+
+    protected Map<String, RecordPath> getRecordPaths(ProcessContext context) {
+        return context.getProperties().keySet()
+            .stream().filter(p -> p.isDynamic() && !p.getName().contains(RECORD_READER.getName()))
+            .collect(Collectors.toMap(
+                e -> e.getName(),
+                e ->  {
+                    String val = context.getProperty(e).getValue();
+                    return cache.getCompiled(val);
+                })
+            );
+    }
+
+    protected Map<String, String> getStats(FlowFile input, Map<String, RecordPath> paths, ProcessContext context, ProcessSession session) {
+        try (InputStream is = session.read(input)) {
+            RecordReaderFactory factory = context.getProperty(RECORD_READER).asControllerService(RecordReaderFactory.class);
+            RecordReader reader = factory.createRecordReader(input, is, getLogger());
+
+            Map<String, Integer> retVal = new HashMap<>();
+            Record record;
+
+            int recordCount = 0;
+            while ((record = reader.nextRecord()) != null) {
+                for (Map.Entry<String, RecordPath> entry : paths.entrySet()) {
+                    RecordPathResult result = entry.getValue().evaluate(record);
+                    Optional<FieldValue> value = result.getSelectedFields().findFirst();
+                    if (value.isPresent() && value.get().getValue() != null) {
+                        FieldValue fieldValue = value.get();
+                        String approxValue = fieldValue.getValue().toString();
+                        String key = String.format("%s.%s", entry.getKey(), approxValue);
+                        Integer stat = retVal.containsKey(key) ? retVal.get(key) : 0;
+                        Integer baseStat = retVal.containsKey(entry.getKey()) ? retVal.get(entry.getKey()) : 0;
+                        stat++;
+                        baseStat++;
+
+                        retVal.put(key, stat);
+                        retVal.put(entry.getKey(), baseStat);
+                    }
+                }
+
+                recordCount++;
+            }
```

Comment:
Magic Strings

Reply:
?

Reply:
"record_count"  => constant field maybe?

Reply:
not a biggie

Reply:
Done.